zerver/tests/test_email_mirror.py

```diff
@@ -637,9 +637,9 @@ def test_receive_stream_email_messages_empty_body(self) -> None:
         incoming_valid_message['To'] = stream_to_address
         incoming_valid_message['Reply-to'] = self.example_email('othello')
 
-        with mock.patch('zerver.lib.email_mirror.logging.warning') as mock_warn:
+        with self.assertLogs(level="WARNING") as m:
             process_message(incoming_valid_message)
-            mock_warn.assert_called_with("Email has no nonempty body sections; ignoring.")
+        self.assertEqual(m.output, ["WARNING:root:Email has no nonempty body sections; ignoring."])
```

Comment:
We should assert that this logging call is made with the top-level logger rather than just checking if for `root`. So, this should have something like `self.assertLogs(logger='zerver.lib.email_mirror', level='WARNING')`. 
Looking at the actual logger call will help as we're making sure that these are called with that https://github.com/zulip/zulip/blob/master/zerver/lib/email_mirror.py#L42

(It probably is better to have logger string `zerver.lib.email_mirror` saved as a variable somewhere in the file and use it rather than typing it everywhere in the file)

Reply:
made this change in [commit](https://github.com/zulip/zulip/pull/15708/commits/b808b2ac7365786345078c4c878cc799467dd8ad) also changed email_mirror.py to use the correct logger.